src/posts/lucky_0.25.cr

```diff
@@ -0,0 +1,370 @@
+
+      private def update_billing
+      end
+
+      private def card_still_valid?
+        false
+      end
+    end
+    ```
+
+    Lastly, there were some instances where you may be updating a record, and expecting a callback when everything was done. However, in the case of
+    a record that doesn't need to update anything, you'd never have any callbacks that fired. (i.e. kick off background job, or email, etc...)
+
+    To help with this, we've added 1 additional callback called `after_completed`. This callback is always called when the `save_status` is set to `:saved`.
+    This means that even if the record never touches the database, the `after_completed` callback will still run.
+
+    ```crystal
+    class SaveUser < User::SaveOperation
+
+      after_commit do |updated_user|
+        # This is only called if something on the user actually changed
+        AfterCommitJob.perform(updated_user.id)
+      end
+
+      after_completed do |updated_user|
+        # This is always called if the user is saved
+        MetricsUpdaterJob.perform(updated_user.id)
+      end
+
+    end
+    ```
+
+    ### `with_defaults` is now `tag_defaults`
+
+    When you create components like `Shared::Field`, you can use the `tag_defaults` method to apply default attributes to a set of HTML elements.
+    This method was called `with_defaults` before, but has been renamed for some clarity on what it does.
+
+    ### Changes to the Query objects
+
+    The `Query` objects also got a few new changes. The main one being that queries no longer mutate the object. Prior to this release, appending any
+    query method would mutate the original object. This made building queries really nice and easy, but presented a problem that many faced. For example:
+
+    ```crystal
+    q = UserQuery.new
+    if current_user.admin?
+      q.admin(true)
+    end
+
+    # this would fail because we've already mutated the query
+    # on the count, and can no longer query with the username ordering
+    user_total_count = q.select_count
+    users = q.username.asc_order
+    ```
+
+    In that case, you had to make sure you cloned the query, and they were in the correct order. Now we no longer mutate the query, but this requires changing your
+    code just slightly.
+
+    ```crystal
+    q = UserQuery.new
+    if current_user.admin?
+      q = q.admin(true)
+    end
+
+    user_total_count = q.select_count
+    users = q.username.asc_order
+    ```
+
+    With this change, this also means that defining default queries in your query class `initialize` need to change. To set a default, you'll now use the `defaults` method.
+
+    ```crystal
+    class UserQuery < User::BaseQuery
+      def initialize
+        defaults &.admin(true)
+      end
+    end
+    ```
+
+    ### Type-safe WHERE "OR"
+
+    This was one of the most requested methods we got! We held off since adding this in makes SQL queries quite a bit more complex, but we finally got it! (**sort of... see note below)
```

Comment:
```suggestion
    This was one of the most requested additions! We held off since adding this in makes SQL queries quite a bit more complex, but we finally got it! (**sort of... see note below)
```